File: generate-icons.js
#!/usr/bin/env node
/**
 * 簡単なプレースホルダーアイコンを生成
 * PNG形式で最小限の有効なファイルを作成
 */
const fs = require('fs');
const path = require('path');

// アイコンサイズ
const sizes = [72, 96, 128, 144, 152, 192, 384, 512];

// テーマカラー #5c6bc0 = rgb(92, 107, 192)
const r = 92, g = 107, b = 192;

function createPNG(width, height, r, g, b) {
  // 最小限の有効なPNGを作成
  const PNG = require('pngjs').PNG;
  const png = new PNG({ width, height });

  // 背景色を設定
  for (let y = 0; y < height; y++) {
    for (let x = 0; x < width; x++) {
      const idx = (width * y + x) << 2;

      // 中央に白い円を描画
      const centerX = width / 2;
      const centerY = height / 2;
      const radius = width * 0.35;
      const distance = Math.sqrt(Math.pow(x - centerX, 2) + Math.pow(y - centerY, 2));

      if (distance < radius) {
        // 白い円の内側
        const innerRadius = width * 0.23;
        if (distance < innerRadius) {
          // 内側の円はテーマカラー
          png.data[idx] = r;
          png.data[idx + 1] = g;
          png.data[idx + 2] = b;
          png.data[idx + 3] = 255;
        } else {
          // 中間は白
          png.data[idx] = 255;
          png.data[idx + 1] = 255;
          png.data[idx + 2] = 255;
          png.data[idx + 3] = 255;
        }
      } else {
        // 背景はテーマカラー
        png.data[idx] = r;
        png.data[idx + 1] = g;
        png.data[idx + 2] = b;
        png.data[idx + 3] = 255;
      }
    }
  }

  return PNG.sync.write(png);
}

// pngjs がインストールされているか確認
try {
  require.resolve('pngjs');

  const iconsDir = path.join(__dirname, 'icons');
  if (!fs.existsSync(iconsDir)) {
    fs.mkdirSync(iconsDir);
  }

  sizes.forEach(size => {
    const buffer = createPNG(size, size, r, g, b);
    const filename = path.join(iconsDir, `icon-${size}.png`);
    fs.writeFileSync(filename, buffer);
    console.log(`✓ 生成完了: icons/icon-${size}.png`);
  });

  console.log('\nすべてのアイコンが生成されました！');
} catch (err) {
  console.log('pngjs がインストールされていません。インストール中...');
  console.log('npm install pngjs を実行してください。');
}

File: generate_icons.py
#!/usr/bin/env python3
"""
簡単なプレースホルダーアイコンを生成するスクリプト
"""
from PIL import Image, ImageDraw, ImageFont
import os

# アイコンサイズ
sizes = [72, 96, 128, 144, 152, 192, 384, 512]

# テーマカラー
bg_color = (92, 107, 192)  # #5c6bc0
text_color = (255, 255, 255)  # white

# iconsディレクトリを確認
icons_dir = "icons"
if not os.path.exists(icons_dir):
    os.makedirs(icons_dir)

for size in sizes:
    # 新しい画像を作成
    img = Image.new('RGB', (size, size), bg_color)
    draw = ImageDraw.Draw(img)

    # 絵文字アイコン風の円を描画（シンプルなデザイン）
    margin = size // 8
    circle_bbox = [margin, margin, size - margin, size - margin]

    # 白い円を描画
    draw.ellipse(circle_bbox, fill=text_color)

    # 内側に小さい円を描画（計算機のイメージ）
    inner_margin = size // 4
    inner_circle_bbox = [inner_margin, inner_margin, size - inner_margin, size - inner_margin]
    draw.ellipse(inner_circle_bbox, fill=bg_color)

    # ファイルを保存
    filename = f"{icons_dir}/icon-{size}.png"
    img.save(filename, "PNG")
    print(f"✓ 生成完了: {filename}")

print("\nすべてのアイコンが生成されました！")
